# 65694 Network error messages are wider than browser window and they look complex and messy when zoom level is bigger than 100%
# 71447 when zooming, text is lost from the right side of columns
--- mozilla/layout/generic/nsGfxScrollFrame.cpp.orig	2007-09-14 11:10:42.000000000 +0300
+++ mozilla/layout/generic/nsGfxScrollFrame.cpp	2007-09-17 14:20:58.000000000 +0300
@@ -72,20 +72,22 @@
 #include "nsDocShellCID.h"
 #include "nsIDOMHTMLDocument.h"
 #include "nsEventDispatcher.h"
 #include "nsContentUtils.h"
 #include "nsLayoutUtils.h"
 #ifdef ACCESSIBILITY
 #include "nsIAccessibilityService.h"
 #endif
 #include "nsDisplayList.h"
 #include "nsBidiUtils.h"
+#include "nsIDocShell.h"
+#include "nsDocShellLoadTypes.h"
 
 //----------------------------------------------------------------------
 
 //----------nsHTMLScrollFrame-------------------------------------------
 
 nsIFrame*
 NS_NewHTMLScrollFrame(nsIPresShell* aPresShell, nsStyleContext* aContext, PRBool aIsRoot)
 {
   return new (aPresShell) nsHTMLScrollFrame(aPresShell, aContext, aIsRoot);
 }
@@ -421,21 +423,29 @@ nsHTMLScrollFrame::ReflowScrolledFrame(c
   nscoord availWidth = aState.mReflowState.ComputedWidth() + paddingLR;
 
   if (aAssumeVScroll) {
     nsSize vScrollbarPrefSize = 
       mInner.mVScrollbarBox->GetPrefSize(NS_CONST_CAST(nsBoxLayoutState&, 
                                                        aState.mBoxState));
     availWidth = PR_MAX(0, availWidth - vScrollbarPrefSize.width);
   }
   // pixel align the content
   nsPresContext* presContext = PresContext();
-  if (presContext->TrueZoomMode() && aFirstPass) {
+  PRUint32 loadType = 0;
+  {
+    nsCOMPtr<nsISupports> container = presContext->GetContainer();
+    nsCOMPtr<nsIDocShell> docShell(do_QueryInterface(container));
+    if (docShell)
+      docShell->GetLoadType(&loadType);
+  }
+  
+  if (presContext->TrueZoomMode() && aFirstPass && loadType != LOAD_ERROR_PAGE && aState.mStyles.mHorizontal != NS_STYLE_OVERFLOW_HIDDEN) {
     availWidth = availWidth * presContext->GetFullZoom();
   }
   nscoord twp = nsPresContext::CSSPixelsToAppUnits(1);
   availWidth -=  availWidth % twp;
 
   if (!aFirstPass)
     mInner.mScrolledFrame->AddStateBits(NS_FRAME_IS_DIRTY);
 
   // We're forcing the padding on our scrolled frame, so let it know what that
   // padding is.
